draft-ietf-quic-version-negotiation.md

```diff
@@ -392,13 +392,32 @@ its length is not divisible by four), then the endpoint MUST close the
 connection; if the connection was using QUIC version 1, that connection closure
 MUST use a transport error of type TRANSPORT_PARAMETER_ERROR. If an endpoint
 receives a Chosen Version equal to zero, or any Available Version equal to zero,
-it MUST treat it as a parsing failure.
+it MUST treat it as a parsing failure. If a server receives a Version
+Information where the Chosen Version is not included in Available Versions, it
+MUST treat it as a parsing failure.
 
 Every QUIC version that supports version negotiation MUST define a method for
 closing the connection with a version negotiation error. For QUIC version 1,
 version negotiation errors are signaled using a transport error of type
 VERSION_NEGOTIATION_ERROR; see {{iana-error}}.
 
+When a server receives a client's first flight, the server will first establish
+which QUIC version is in use for this connection in order to properly parse the
+first flight. For example, the server determines that QUIC version 1 is in use
+by observing that the Version field of the first Long Header packet it receives
+is set to 0x00000001. When the server then parses the client's Version
```

Comment:
I'm concerned about this "for example". Suppose we defined a QUIC v3 which used the same version field. In S 2.2. we said this was version specific and it has to be for any *pair* of versions.

Reply:
The QUIC version field contains the QUIC version, so I'm not sure I follow how QUIC version 3 would use the version field of 1.

Reply:
NM. I misread this as the client determining the server version. I agree this text is fine.

Reply:
I think that @ekr is implicitly referring to the potential for the version field to become ossified where we are forced to identify v3 in some other way.